src/project-manager.ts

```diff
@@ -891,18 +889,24 @@ export class ProjectConfiguration {
 			return;
 		}
 
-		// paths have been forward-slashified by TS (see toUnixPath below)
-
+		// Add global declaration files from the whole workspace
 		for (const uri of this.fs.uris()) {
 			const fileName = uri2path(uri);
-			if (isGlobalTSFile(fileName) ||
-				(isDeclarationFile(fileName) && this.expectedFilePaths.includes(toUnixPath(fileName)))) {
+			if (isGlobalTSFile(fileName)) {
 				const sourceFile = program.getSourceFile(fileName);
 				if (!sourceFile) {
 					this.getHost().addFile(fileName);
 				}
 			}
 		}
+
+		// Add files that belong to the project according to tsconfig.json settings
```

Comment:
This changes the previous behaviour of only adding declaration files project-wide and adding regular TS files as-needed. Are you sure this is what you want?